toolbox/library/src/OSQP.cpp

```diff
@@ -0,0 +1,589 @@
+        return false;
+    }
+
+    // INPUTS
+    // ======
+    //
+    // 1) Hessian Matrix (nV x nV)
+    // 2) Gradient vector (1 x nV)
+    // 3) Optional: Constraints matrix (nC x nV)
+    // 4) Optional: Constraints lower bounds (1 x nV)
+    // 5) Optional: Constraints upper bounds (1 x nV)
+    // 6) Optional: Variables lower bounds (1 x nC)
+    // 7) Optional: Variables upper bounds (1 x nC)
+    //
+    // OUTPUT
+    // ======
+    //
+    // 1) Primal solution (1 x nV)
+    // 2) Status of the qp solver (1x1)
+    // 3) Optional: Value of the object function (1x1)
+
+    InputPortsInfo inputPortsInfo;
+    OutputPortsInfo outputPortsInfo;
+
+    // Inputs
+    inputPortsInfo.push_back({InputIndex::Hessian,
+                              Port::Dimensions{Port::DynamicSize, Port::DynamicSize},
+                              Port::DataType::DOUBLE});
+    inputPortsInfo.push_back(
+        {InputIndex::Gradient, Port::Dimensions{Port::DynamicSize}, Port::DataType::DOUBLE});
+    outputPortsInfo.push_back(
+        {OutputIndex::PrimalSolution, Port::Dimensions{Port::DynamicSize}, Port::DataType::DOUBLE});
+    outputPortsInfo.push_back({OutputIndex::Status, Port::Dimensions{1}, Port::DataType::DOUBLE});
+
+    // Optional inputs
+    size_t numberOfInputs = InputIndex::Gradient;
+    if (useLbA || useUbA) {
+        InputIndex_constraints = ++numberOfInputs;
+        inputPortsInfo.push_back({InputIndex_constraints,
+                                  Port::Dimensions{Port::DynamicSize, Port::DynamicSize},
+                                  Port::DataType::DOUBLE});
+    }
+    if (useLbA) {
+        InputIndex_lbA = ++numberOfInputs;
+        inputPortsInfo.push_back(
+            {InputIndex_lbA, Port::Dimensions{Port::DynamicSize}, Port::DataType::DOUBLE});
+    }
+    if (useUbA) {
+        InputIndex_ubA = ++numberOfInputs;
+        inputPortsInfo.push_back(
+            {InputIndex_ubA, Port::Dimensions{Port::DynamicSize}, Port::DataType::DOUBLE});
+    }
+    if (useLb) {
+        InputIndex_lb = ++numberOfInputs;
+        inputPortsInfo.push_back(
+            {InputIndex_lb, Port::Dimensions{Port::DynamicSize}, Port::DataType::DOUBLE});
+    }
+    if (useUb) {
+        InputIndex_ub = ++numberOfInputs;
+        inputPortsInfo.push_back(
+            {InputIndex_ub, Port::Dimensions{Port::DynamicSize}, Port::DataType::DOUBLE});
+    }
+
+    // Optional outputs
+    size_t numberOfOutputs = OutputIndex::Status;
+    if (computeObjVal) {
+        OutputIndex_objVal = ++numberOfOutputs;
+        outputPortsInfo.push_back(
+            {OutputIndex_objVal, Port::Dimensions{1}, Port::DataType::DOUBLE});
+    }
+
+    if (!blockInfo->setPortsInfo(inputPortsInfo, outputPortsInfo)) {
+        bfError << "Failed to configure input / output ports.";
+        return false;
+    }
+
+    return true;
+}
+
+bool wbt::block::OSQP::solverInitialization(const BlockInformation*blockInfo)
```

Comment:
Missing space in parameter. Did your IDE pick clang-format?

Reply:
To be honest, I have to check how to do that on Windows, and who can provide clang-format there.

Reply:
I think that `clang-tools` in conda-forge should have clang-format also on Windows.